src/check.c

```diff
@@ -272,6 +272,8 @@ static void auto_rename(DOS_FILE * file)
 	sprintf(num, "%07lu", (unsigned long)number);
 	memcpy(file->dir_ent.name, "FSCK", 4);
 	memcpy(file->dir_ent.name + 4, num, 7);
+        if (file->lfn)
+           lfn_remove(file->lfn_offset, file->offset);
```

Comment:
You don't clear the `file->lfn` field. This is not a problem in the other case where it is called since there the whole file is dropped. Here it should be updated.